sqlglot/dialects/teradata.py

```diff
@@ -13,6 +13,29 @@
 from sqlglot.tokens import TokenType
 
 
+def _date_add_sql(
+    kind: t.Literal["+", "-"],
+) -> t.Callable[[Teradata.Generator, exp.DateAdd | exp.DateSub], str]:
+    def func(self: Teradata.Generator, expression: exp.DateAdd | exp.DateSub) -> str:
+        this = self.sql(expression, "this")
+        unit = expression.args.get("unit")
+
+        if isinstance(expression.expression, exp.Neg):
```

Comment:
what happens if there are multiple negatives? maybe it's better to simplify first, and add a method to expression which is like is_negative or something which handles the case? i can add this

Reply:
https://github.com/tobymao/sqlglot/commit/2ea438b89f76a357390d657fe3f9e01d2a79e7e4

Reply:
thanks for adding that! I updated to check `if expression.expression.is_negative`